src/audio/mux/mux.c

```diff
@@ -369,7 +369,8 @@ static int mux_copy(struct comp_dev *dev)
 	sink_bytes = frames * buffer_frame_bytes(sink);
 
 	/* produce output */
-	cd->mux(dev, sink, &sources[0], frames, &cd->config.streams[0]);
+	cd->mux(dev, sink, (const struct comp_buffer **)&sources[0],
```

Comment:
Instead of casting here the declaration should have const in `struct comp_buffer *source;` variable declaration (if checkpatch does not complain)